### ISSUE TITLE
Allow more friendly text selection to clipboard

### ISSUE THREAD

**Issue by Friptick** [wustho/epy#77]
Selecting and copying text crom `epy` to the clipboard is not a great user experience. It is only possible with the GUI (mouse) pointer, and the resulting string will be full of newlines and whitespace (from the left and right margins). Which makes a mess when pasted.

It would be great to have some kind of select-to-clipboard feature. Ideally keyboard-based but at least something that strips all the excess whitespace before sending it to the clipboard.

Thanks for considering and for your work on this project.

**Comment by newptcai** [comment]
I use kitty terminal emulator. It has good support for selecting text from the screen.

**Comment by Friptick** [comment]
Yes, I use Kitty too, and the problem described is still a problem.

**Comment by newptcai** [comment]
Sorry for any earlier lack of clarity.

In Kitty, you can configure a custom pager to view your terminal's scrollback history in Neovim.

Here's how I have it set up:

### 1. Create a `pager.sh` Script

Place the following script in `~/.config/kitty/pager.sh`:

```bash
#!/usr/bin/env bash
set -eu

if [ "$#" -eq 3 ]; then
    INPUT_LINE_NUMBER=${1:-0}
    CURSOR_LINE=${2:-1}
    CURSOR_COLUMN=${3:-1}
    AUTOCMD_TERMCLOSE_CMD="call cursor(max([0,${INPUT_LINE_NUMBER}-1])+${CURSOR_LINE}, ${CURSOR_COLUMN})"
else
    AUTOCMD_TERMCLOSE_CMD="normal G"
fi

exec nvim 63<&0 0</dev/null \
    -u NONE \
    -c "map <silent> q :qa!<CR>" \
    -c "set shell=bash scrollback=100000 termguicolors laststatus=0 clipboard+=unnamedplus" \
    -c "autocmd TermEnter * stopinsert" \
    -c "autocmd TermClose * ${AUTOCMD_TERMCLOSE_CMD}" \
    -c 'terminal sed </dev/fd/63 -e "s/'$'\x1b'']8;;file:[^\]*[\]//g" && sleep 0.01 && printf "'$'\x1b'']2;"'
```


This script configures Neovim to display the scrollback buffer with syntax highlighting and proper cursor positioning.

### 2. Update `kitty.conf`

In the same directory (`~/.config/kitty/`), add the following line to your `kitty.conf`:

```
# Default scrollback pager
# See https://github.com/kovidgoyal/kitty/issues/719#issuecomment-952039731
scrollback_pager ~/.config/kitty/pager.sh 'INPUT_LINE_NUMBER' 'CURSOR_LINE' 'CURSOR_COLUMN'
```


This configuration tells Kitty to use your custom script as the pager when you press `Ctrl+Shift+h`.

### 3. Usage

After setting this up, pressing `Ctrl+Shift+h` in Kitty will open the scrollback buffer in Neovim. You can navigate and copy text as you would in Vim. For instance, to copy a specific section of text, you can use Neovim's [Visual Block Mode](https://builtin.com/articles/vim-visual-mode).

This setup allows you to seamlessly interact with your terminal's history using the powerful editing capabilities of Neovim.

**Comment by Friptick** [comment]
>It is only possible with the GUI (mouse) pointer, and the resulting string will be full of newlines and whitespace (from the left and right margins). Which makes a mess when pasted.

For the record, the whitespace issue can be fixed (minimally) by doing `wl-copy $(printf '%s' "$(wl-paste)")` (assuming Wayland) straight after copying with the mouse.

**Comment by Friptick** [comment]
The solution proposed by @newptcai (thanks) involving the scrollback buffer does indeed look optimal. Best to delegate functionality like this to the terminal rather than duplicate it in the app as I originally proposed.

Closing.
